dask/order.py

```diff
@@ -109,7 +109,10 @@ def dependents_key(x):
     i = 0
 
     stack = [k for k, v in dependents.items() if not v]
-    stack = sorted(stack, key=dependencies_key)
+    if len(stack) < 10000:
```

Comment:
How did you come up with these numbers?

Reply:
Totally arbitrary

Reply:
In practice numbers around 1-16 seem to be common and good to sort.  Numbers like 100k seem to be bad to sort.  I chose something in the middle.

Reply:
Fine by me. Avoiding sorting in expensive situations seems fine, and the numbers are sufficiently high that they're unlikely to affect complicated graph structures that might benefit more from sorting.